sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2671,6 +2671,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val ANALYZE_PARTITION_STATS_ENABLED =
+    buildConf("spark.sql.statistics.update.partitionStats.enabled")
```

Comment:
1. Do we have other configurations under `spark.sql.statistics.update.*`? We should not add a new namespace for a single configuration. Shall we rename like the following?
```
- spark.sql.statistics.update.partitionStats.enabled
+ spark.sql.statistics.updatePartitionStats.enabled
```

2. Also, the current config name looks too general. Can we revise the config name to give some idea about `ANALYZE TABLE` syntax?

Reply:
Hmm, how about `spark.sql.statistics.updatePartitionStatsInAnalyzeTable.enabled`? it is a bit long but can fully expression the intention 😂

Reply:
+1 for the new name.

Reply:
Thanks, updated.